### PR TITLE
Fixed federated athentication support

### PR DESCRIPTION
**Pull request by Kanapriya**

### CODE REVIEW

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 1]
```diff
@@ -257,35 +254,58 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
...
             }
 
             // Get claim values for the user
-            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            UserStoreManager userStoreManager = null;
+            boolean localAuthentication = false;
+            try {
+                userStoreManager = userRealm.getUserStoreManager();
+                localAuthentication = userStoreManager.isExistingUser(username);
+            } catch (Exception e) {
```
Comment (RKathees) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java]: Here, catch UserStoreException specific exception.

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 2]
```diff
@@ -257,35 +254,58 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
...
+                // User came from federated authentciation
```
Comment (RKathees) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java:271]: Handle the exception properly.

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 3]
```diff
@@ -257,35 +254,60 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
...
-            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            UserStoreManager userStoreManager = null;
+            boolean localAuthentication = false;
+            try {
+                userStoreManager = userRealm.getUserStoreManager();
+                localAuthentication = userStoreManager.isExistingUser(username);
+            }
+            catch (UserStoreException e) {
+                // User came from federated authentciation
+                log.warn ("Error while retrieving the user from federated authentication, e");
```
Comment (pamod) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java:272]: is there any reason for having a warn log ? also should we throw an identity exception here ?

Comment (Kanapriya) [reply components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java:272]: I have changed the warn log to error  and added the UserStore exception to increase the severity of the error. Please find the fixed in following commit
7042a9def49a60d0d96f05a107e7d771566369d6

**Pull request merged**